bundle/runtime-schemas/defs-plugins.json

```diff
@@ -196,19 +196,41 @@
                         "dnsmasq": {
                             "type": "boolean"
                         },
-                        "holes": {
-                            "type": "array",
-                            "items": {
-                                "type": "object",
-                                "required": [
-                                    "port"
-                                ],
-                                "properties": {
-                                    "port": {
-                                        "$ref": "defs.json#/definitions/uint16"
-                                    },
-                                    "protocol": {
-                                        "type": "string"
+                        "portForwarding": {
+                            "type": "object",
+                            "properties": {
+                                "hostToContainer": {
+                                    "type": "array",
```

Comment:
Minor: Definition of "array of objects port+protocol" can be named better, stored and used twice in "hostToContainer" and "containerToHost". To see how it is done you could check i.e. "defs.json#/definitions/uint16"

Reply:
I agree, this would be a good upgrade. I've just tried to do that though and it seems like there's a bug in libocispec, where if you try to add an array with defs in that way, when using the types as reference do not get the suffix "_element", causing buildtime errors. Without using defs, the suffix is added correctly. This can be seen functioning with the "loopback" element as well.